### ISSUE TITLE
CVarArgList not working correctly on macOS M1

### ISSUE THREAD

**Issue by ekrich** [scala-native/scala-native#2494]
Multiple tests fail on Scala 3 and Scala 2.13.

Summary:

- Floats and Doubles don't work.
- All tests with greater than 6 args don't work.
- Mixed args don't work.

Here are some samples.
```scala
[error] Test scala.scalanative.unsafe.CVarArgListTest.mixArgs7 failed: java.lang.AssertionError: 1 2 3 4 5 6 -1717986918 0.0 2.2 0.0 3.3 0.0 4.4 0.0 != 1 2 3 4 5 6 7 1.1 2.2 3.3 4.4 5.5 6.6 7.7, took 0.000 sec
[error] Test scala.scalanative.unsafe.CVarArgListTest.doubleArgs3 failed: java.lang.AssertionError: 0.0 0.0 0.0 != 1.1 2.2 3.3, took 0.000 sec
[error] Test scala.scalanative.unsafe.CVarArgListTest.floatArgs9 failed: java.lang.AssertionError: 0.0 0.0 0.0 0.0 0.0 0.0 1.1 0.0 2.2 != 1.1 2.2 3.3 4.4 5.5 6.6 7.7 8.8 9.9, took 0.000 sec
[error] Test scala.scalanative.unsafe.CVarArgListTest.ushortArgs7 failed: java.lang.AssertionError: 1 2 3 4 5 6 0 != 1 2 3 4 5 6 7, took 0.000 sec
[info] Test run finished: 51 failed, 0 ignored, 127 total, 0.083s
```

**Comment by WojciechMazur** [comment]
@ekrich can you try to rerun the tests. I think that it should be fixed recently in #2573. The fix there was mainly targeted for Linux based Arm64, but some of the changes should also affect macOS.

**Comment by ekrich** [comment]
@WojciechMazur I just included the whole output to make it easier - it is pretty long but we have lots of tests which is a good thing.
[M1.txt](https://github.com/scala-native/scala-native/files/8142851/M1.txt)

**Comment by WojciechMazur** [comment]
Thank you Eric, I hoper that it was already fixed. Anyway, I'm going to set up an instance of Apple M1 as a Service in one of the cloud providers I've found and fix that this weekend. I really would like to have it fixed for 0.4.4

**Comment by ekrich** [comment]
Sorry, I haven't had more time to help out.
